node/middlewares/internals/category.ts

```diff
@@ -12,86 +12,47 @@ import { slugify } from '../../utils/slugify'
 
 type CategoryTypes = 'DEPARTMENT' | 'CATEGORY' | 'SUBCATEGORY'
 
-interface IdentifiedCategory {
-  type: CategoryTypes
-  map: string
-  id: string
-  params: {
-    department?: string
-    category?: string
-    subcategory?: string
-    terms?: string
-  }
-  isActive: boolean
+interface IdentifiedCategory extends Category {
+  parentsNames: string[]
 }
 
-const categoriesFromCategoryTree = async (
-  category: Category,
-  ctx: Context
-): Promise<IdentifiedCategory[]> => {
-  const { catalogGraphQL } = ctx.clients
-  const { parentCategoryId, name } = category
-
-  if (!parentCategoryId) {
-    const identifiedCategory: IdentifiedCategory = {
-      id: category.id,
-      isActive: category.isActive,
-      map: 'c',
-      params: {
-        department: slugify(name),
-      },
-      type: 'DEPARTMENT',
-    }
-    return [identifiedCategory]
-  }
-
-  const parentCategory = await catalogGraphQL
-    .category(parentCategoryId)
-    .then(res => res!.category)
-
-  const identifiedCategories = await categoriesFromCategoryTree(
-    parentCategory,
-    ctx
-  )
-  const { type, params, map } = identifiedCategories[0]
-
-  if (type === 'DEPARTMENT') {
-    const identifiedCategory: IdentifiedCategory = {
-      id: category.id,
-      isActive: category.isActive,
-      map: `${map},c`,
-      params: {
-        ...params,
-        category: slugify(name),
-      },
-      type: 'CATEGORY',
-    }
-    return [identifiedCategory, ...identifiedCategories]
-  }
-  if (type === 'CATEGORY') {
-    const identifiedCategory: IdentifiedCategory = {
-      id: category.id,
-      isActive: category.isActive,
-      map: `${map},c`,
-      params: {
-        ...params,
-        subcategory: slugify(name),
-      },
-      type: 'SUBCATEGORY',
-    }
-    return [identifiedCategory, ...identifiedCategories]
-  }
-  const identifiedCategory = {
-    id: category.id,
-    isActive: category.isActive,
-    map: `${map},c`,
-    params: {
-      ...params,
-      terms: params.terms ? `${params.terms}${slugify(name)}` : slugify(name),
-    },
-    type,
+const getCategoryType = (
+  parentsNames: string[]
```

Comment:
This variable contains not only the parent names as well as the current categories. I'd rather call it categoryTree to avoid confusion